js/infobar.js

```diff
@@ -125,35 +131,74 @@
      * @param {string} pageLang - The current page language
      */
     InfoBar.prototype.getOfferedLang = function(acceptLangs, pageLang) {
-        var shortUserLang = '';
-        var offeredLang;
+        var langCode = '';
+        var indexMatch = false;
+        var langCodeMatch = false;
+        var offeredLang = false;
 
-        config.lang = pageLang || config.lang;
+        config.pageLang = pageLang || config.pageLang;
         config.availableLangs = InfoBar.prototype.getAvailableLangs();
 
-        // If there are no available languages, we have nothing to do.
+        // If there are no available languages, we are done.
         if (!config.availableLangs) {
             return false;
         }
 
-        // If the users accept language(s) match page lang, we are done.
-        if (InfoBar.prototype.userLangPageLangMatch(acceptLangs)) {
+        // If the page language is the user's primary language, we are done.
+        if (config.pageLang === acceptLangs[0]) {
             return false;
         }
 
         // Compare the user's accept languages against available
         // languages to find the best language
-        $.each(acceptLangs, function(index, userLang) {
+        $.each(acceptLangs, function(index, value) {
+            langCode = value.split('-')[0];
+            indexMatch = $.inArray(value, config.availableLangs) > -1;
+            langCodeMatch = $.inArray(langCode, config.availableLangs) > -1;
+
+            if (indexMatch || langCodeMatch) {
+                offeredLang = acceptLangs[index];
+                return;
```

Comment:
I believe you need to explicitly `return false;` to break out of a jQuery `$.each` loop.